### PR TITLE
[RLlib] Replace "seq_lens" w/ SampleBatch.SEQ_LENS.

### PR DESCRIPTION
**Pull request by sven1977**
- Replaces "seq_lens" w/ SampleBatch.SEQ_LENS.
- Adds SampleBatch.T, SampleBatch.ENV_ID, SampleBatch.UNROLL_ID fields (used to be done as plain strings, which is error prone).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_advanced.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 import ray._private.profiling as profiling
 from ray._private.test_utils import (client_test_enabled,
-                                     RayTestTimeoutException, SignalActor)
+                                     RayTestTimeoutException)
```
Comment (sven1977) [inline python/ray/tests/test_advanced.py:18]: Not used -> fixes LINT error.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Hey Sven, I think all changes look good!

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Hey Sven, I think all changes look good.

**Comment by sven1977** [comment]
Thanks @ArturNiederfahrenhorst ! :)

**Pull request merged**